core/src/test/scala/io/projectglow/vcf/VCFFileWriterSuite.scala

```diff
@@ -464,6 +467,53 @@ class SingleFileVCFWriterSuite extends VCFFileWriterSuite("bigvcf") {
     val rereadDs = spark.read.format("vcf").load(tempFile)
     assert(rereadDs.sort("start").collect sameElements ds.sort("start").collect)
   }
+
+  test("Bigvcf header check for empty file with determined header") {
+    val sess = spark
+    import sess.implicits._
+
+    val tempFile = createTempVcf.toString
+
+    spark
+      .sparkContext
+      .emptyRDD[VCFRow]
+      .toDS
+      .repartition(1)
+      .write
+      .option("vcfHeader", NA12878)
+      .format(sourceName)
+      .save(tempFile)
+
+    val rewrittenDs = spark
+      .read
+      .format(readSourceName)
+      .option("vcfRowSchema", true)
+      .load(tempFile)
+
+    val truthHeader = VCFMetadataLoader.readVcfHeader(sparkContext.hadoopConfiguration, NA12878)
+    val writtenHeader = VCFMetadataLoader.readVcfHeader(sparkContext.hadoopConfiguration, tempFile)
+
+    assert(truthHeader.getMetaDataInInputOrder.equals(writtenHeader.getMetaDataInInputOrder))
```

Comment:
One more check: can you ensure the sample match? (`truthHeader.getGenotypeSamples == writtenHeader.getGenotypeSamples`)

Reply:
Added